core/src/main/java/org/apache/gravitino/job/local/LocalJobExecutor.java

```diff
@@ -156,12 +156,12 @@ public String submitJob(JobTemplate jobTemplate) {
     String newJobId = LOCAL_JOB_PREFIX + UUID.randomUUID();
     Pair<String, JobTemplate> jobPair = Pair.of(newJobId, jobTemplate);
 
-    // Add the job template to the waiting queue
-    if (!waitingQueue.offer(jobPair)) {
-      throw new IllegalStateException("Waiting queue is full, cannot submit job: " + jobTemplate);
-    }
-
     synchronized (lock) {
```

Comment:
Does this operation need to be protected by the lock?

Reply:
Which operation?